src/Core/Billing/TrialInitiation/Registration/Implementations/SendTrialInitiationEmailForRegistrationCommand.cs

```diff
@@ -0,0 +1,71 @@
+﻿#nullable enable
+using Bit.Core.Auth.Models.Business.Tokenables;
+using Bit.Core.Billing.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Bit.Core.Tokens;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.Billing.TrialInitiation.Registration.Implementations;
+
+public class SendTrialInitiationEmailForRegistrationCommand(
+    IUserRepository userRepository,
+    GlobalSettings globalSettings,
+    IMailService mailService,
+    IDataProtectorTokenFactory<RegistrationEmailVerificationTokenable> tokenDataFactory)
+    : ISendTrialInitiationEmailForRegistrationCommand
+{
+    public async Task<string?> Handle(
+        string email,
+        string? name,
+        bool receiveMarketingEmails,
+        ProductTierType productTier,
+        IEnumerable<ProductType> products)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(email, nameof(email));
+
+        var userExists = await CheckUserExistsConstantTimeAsync(email);
+        var token = GenerateToken(email, name, receiveMarketingEmails);
+
+        if (!globalSettings.EnableEmailVerification)
+        {
+            await PerformConstantTimeOperationsAsync();
```

Comment:
❓ Not a blocker, just curious what the `Task.Delay` is for?

Reply:
Good question, I just pushed a change adding an explanatory comment. The idea is to mitigate timing attacks where an attacker can detect if a user exists based on how long the request takes to complete